homeassistant/components/mill/climate.py

```diff
@@ -84,6 +93,25 @@ async def set_room_temp(service: ServiceCall) -> None:
         DOMAIN, SERVICE_SET_ROOM_TEMP, set_room_temp, schema=SET_ROOM_TEMP_SCHEMA
     )
 
+    async def max_heating_power(service: ServiceCall) -> None:
+        """Limit heating power."""
+        entity_id = service.data.get(ATTR_ENTITY_ID)
+        heating_power = service.data.get(ATTR_MAX_HEATING_POWER)
+        for entity in entities:
+            if entity.entity_id == entity_id:
+                await mill_data_coordinator.mill_data_connection.max_heating_power(
+                    entity.heater_id, heating_power
+                )
+                return
+        raise ValueError(f"Entity id {entity_id} not found")
+
+    hass.services.async_register(
```

Comment:
We should only register services in `async_setup` unless they are entity services (which might work here as well)